src/core/control/tools/StrokeHandler.cpp

```diff
@@ -266,9 +267,20 @@ void StrokeHandler::onButtonReleaseEvent(const PositionInputData& pos, double zo
     Layer* layer = page->getSelectedLayer();
 
     UndoRedoHandler* undo = control->getUndoRedoHandler();
-
     undo->addUndoAction(std::make_unique<InsertUndoAction>(page, layer, stroke.get()));
 
+    // If stroke was on last page, ensure there is always an empty last page by creating one (if configured)
+    auto* doc = control->getDocument();
+    if (!doc->isAttachPdf() && settings->getAddEmptyLastPage()) {
+        doc->lock();
+        auto currentPage = control->getCurrentPageNo();
+        auto totalPages = doc->getPageCount() - 1;
+        doc->unlock();
+        if (currentPage == totalPages) {
```

Comment:
The naming is a bit misleading. 
```suggestion
        auto lastPage = doc->getPageCount() - 1;
        doc->unlock();
        if (currentPage == lastPage) {
```